extensions/agroal/runtime/src/main/java/io/quarkus/agroal/runtime/DataSourceBuildTimeConfig.java

```diff
@@ -15,11 +16,11 @@
     public Optional<String> driver;
 
     /**
-     * Whether we want to use XA.
+     * Whether we want to use regular JDBC transactions, XA, or disable all transactional capabilities.
      * <p>
-     * If used, the driver has to support it.
+     * When enabling XA you will need a driver implementing {@link javax.sql.XADataSource}.
      */
-    @ConfigItem(defaultValue = "false")
-    public boolean xa;
+    @ConfigItem(defaultValue = "ENABLED")
+    public Optional<TransactionIntegration> transactions;
```

Comment:
It shouldn't be an `Optional` if it has a default value as it will never be empty.